cmd/cleanup-controller/main.go

```diff
@@ -116,37 +124,78 @@ func main() {
 				),
 				certmanager.Workers,
 			)
-			webhookController := internal.NewController(
-				webhookControllerName,
+			policyValidatingWebhookController := internal.NewController(
+				policyWebhookControllerName,
 				genericwebhookcontroller.NewController(
-					webhookControllerName,
+					policyWebhookControllerName,
 					setup.KubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations(),
 					kubeInformer.Admissionregistration().V1().ValidatingWebhookConfigurations(),
 					caSecret,
 					config.CleanupValidatingWebhookConfigurationName,
 					config.CleanupValidatingWebhookServicePath,
 					serverIP,
 					int32(servicePort),
-					[]admissionregistrationv1.RuleWithOperations{{
-						Rule: admissionregistrationv1.Rule{
-							APIGroups:   []string{"kyverno.io"},
-							APIVersions: []string{"v2alpha1"},
-							Resources: []string{
-								"cleanuppolicies/*",
-								"clustercleanuppolicies/*",
+					nil,
+					[]admissionregistrationv1.RuleWithOperations{
+						{
+							Rule: admissionregistrationv1.Rule{
+								APIGroups:   []string{"kyverno.io"},
+								APIVersions: []string{"v2alpha1"},
+								Resources: []string{
+									"cleanuppolicies/*",
+									"clustercleanuppolicies/*",
+								},
+							},
+							Operations: []admissionregistrationv1.OperationType{
+								admissionregistrationv1.Create,
+								admissionregistrationv1.Update,
 							},
 						},
-						Operations: []admissionregistrationv1.OperationType{
-							admissionregistrationv1.Create,
-							admissionregistrationv1.Update,
-						},
-					}},
+					},
 					genericwebhookcontroller.Fail,
 					genericwebhookcontroller.None,
 					setup.Configuration,
 				),
 				webhookWorkers,
 			)
+			ttlWebhookController := internal.NewController(
+				ttlWebhookControllerName,
+				genericwebhookcontroller.NewController(
+					ttlWebhookControllerName,
+					setup.KubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations(),
+					kubeInformer.Admissionregistration().V1().ValidatingWebhookConfigurations(),
+					caSecret,
+					config.TtlValidatingWebhookConfigurationName,
+					config.TtlValidatingWebhookServicePath,
+					serverIP,
+					int32(servicePort),
+					&metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      kyverno.LabelCleanupTtl,
+								Operator: metav1.LabelSelectorOpExists,
+							},
+						},
+					},
+					[]admissionregistrationv1.RuleWithOperations{
+						{
+							Rule: admissionregistrationv1.Rule{
+								APIGroups:   []string{"*"},
+								APIVersions: []string{"*"},
+								Resources:   []string{"*"},
+							},
```

Comment:
Do we need to register wildcard to receive all resources? Why not use `objectSelector`?

Reply:
There's an object selector:
```
					&metav1.LabelSelector{
						MatchExpressions: []metav1.LabelSelectorRequirement{
							{
								Key:      kyverno.LabelCleanupTtl,
								Operator: metav1.LabelSelectorOpExists,
							},
						},
					},
```

Reply:
Then is the rule required here?

Reply:
> Then is the rule required here?

Yes, the rules define the resources considered by the webhook.

Reply:
I mean, no rule, no webhook invocation.